flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/Input.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.io;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.runtime.io.AsyncDataInput;
+import org.apache.flink.streaming.runtime.streamrecord.StreamElement;
+
+import java.io.Closeable;
+
+/**
+ * Basic interface for inputs of stream operators.
+ */
+@Internal
+public interface Input extends AsyncDataInput<StreamElement>, Closeable {
```

Comment:
Maybe something along the line of `StreamTask[Async][Data]Input`. This should make role of this interface a lot clearer than the super generic name that it has now.

Reply:
`StreamTaskInput` is not that bad. And the concrete class `StreamTaskNetworkInput`?

Reply:
👍 That sounds likw two names that create a mental model in my head about what this does.

Reply:
@sunhaibotb, are you fine with renaming those renames? (and fyi, since renaming will affect a wip `StreamSelectableTwoInputStreamProcessor` PR.

Reply:
@pnowojski @sunhaibotb ping so that we can address this if we aggree upon my suggestion.

Reply:
> @sunhaibotb, are you fine with renaming those renames? (and fyi, since renaming will affect a wip `StreamSelectableTwoInputStreamProcessor` PR.

It's fine to me. @pnowojski @StefanRRichter

Reply:
I've renamed the classes.